website/content/api-docs/system/managed-keys.mdx

```diff
@@ -0,0 +1,194 @@
+  granting usage.
+
+#### PKCS#11 backend Parameters
+- `type` `(string: "pkcs11")` - To select a PKCS#11 backend, the type parameter must be set to `pkcs11`
+
+- `library` `(string: <required>)` - The name of the `kms_library` stanza to use from Vault's config to
+   lookup the local library path. See [kms_library stanza](/docs/configuration/kms-library) for further details.
+
+- `key_label` `(string: <required>)`: The label of the key to use. If the key
+  does not exist and generation is enabled, this is the label that will be given
+  to the generated key.
+
+- `mechanism` `(string: <required>)` - The encryption/decryption mechanism to use,
+  specified as a hexadecimal (prefixed by 0x) string. The following are supported mechanisms
+    - `0x1041` `CKM_ECDSA`
+    - `0x000D` `CKM_RSA_PKCS_PSS`
+    - `0x0009` `CKM_RSA_PKCS_OAEP`
+    - `0x0001` `CKM_RSA_PKCS`
+
+- `pin` `(string: <required>)`: The PIN for login.
+
+- `slot` `(string: <slot or token label required>)`: The slot number to use,
+  specified as a string (e.g. `"2305843009213693953"`).
+
+- `token_label` `(string: <slot or token label required>)`: The slot token label to
+  use.
+
+- `curve` `(string: "")` - supplies the curve value when using the `CKM_ECDSA`
+  mechanism. This key is required if  The supported values are
+    - `P384`
+    - `P521`
+    - `P256`
+
+- `key_bits` `(int: "")` - supplies the key bits when using `CKM_RSA_PKCS_PSS`,
+  `CKM_RSA_PKCS_OAEP` or `CKM_RSA_PKCS` mechanisms, supported values are
+    - 2048
+    - 3072
+    - 4096
+
+- `force_rw_session` `(string: "false")`: Force all operations to open up
+  a read-write session to the HSM. This is a boolean expressed as a string (e.g.
+  `"true"`). This key is mainly to work around a limitation within AWS's CloudHSM v5
+  pkcs11 implementation.
+
+## Read a Managed Key's properties
+
+This endpoint returns the configuration of the managed key at the given path.
+
+| Method | Path                            |
+|:-------|:--------------------------------|
+| `GET`  | `/sys/managed-keys/:type/:name` |
+
+### Parameters
+- `name` `(string: <required>)` - A unique lowercase name that serves as identifying the key. Must be
+  unique throughout the namespace.
+
+- `type` `(string: <required>)` – The backend type that will be leveraged for the managed key.
+  Supported options are `pkcs11`.
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    http://127.0.0.1:8200/v1/sys/managed-keys/pkcs11/hsm-key1
+```
+
+### Sample Response
+
+```json
+{
+  "UUID": "af0a688e-d2c1-fc07-b365-40325674114d",
+  "allow_generate_key": true,
+  "allow_replace_key": false,
+  "allow_store_key": false,
+  "any_mount": false,
+  "curve": "",
+  "force_rw_session": false,
+  "key_bits": 0,
+  "key_label": "test-kms-root",
+  "library": "softhsm",
+  "mechanism": 1,
+  "name": "hsm-key1",
+  "pin": "redacted",
+  "slot": 1,
+  "token_label": "",
```

Comment:
Should we have an omittempty in the JSON for some of these so they don't show up if empty?

Reply:
I actually prefer that we keep them in the output when empty as it helps document the available options.

Reply:
Fair enough.